### DISCUSSION TITLE
Mesh Augmentations/Transforms: Edge Flipping / Vertex Shifting

### DISCUSSION THREAD

**Discussion by MatteoGioia** [pyg-team/pytorch_geometric#7667] [Q&A]
Hi, I am trying to implement Edge Flipping / Vertex Shifting on triangular meshes.

The idea is shown in this picture ![](https://www.startpage.com/av/proxy-image?piurl=https%3A%2F%2Fwww.pngkit.com%2Fpng%2Fdetail%2F98-986388_mesh-flip-blender-flip-edge.png&sp=1688058751T9c2f210f7bde2fcf7591ca7eb065362760c36251ab27dad5c5dd7eb99c46484d)

In the case of meshes one would also have to consider the angle of each face to avoid incorrectly morphing the original mesh (i.e. only flip edges for flat adjacent faces).

I saw that an this PR #2293 mentions an architecture that has already implemented them (MeshCNN), but the code provided is incomplete/not merged. 

Any pointers on where I could start to implement them?

**Comment by rusty1s** [comment]
I am sorry for the super late reply. In general, I have no objection to integrate better mesh support into PyG. Would it be sufficient to integrate these as transformations, similar to other ones such as `GenerateMeshNormals`, `FaceToEdge`, etc?

**Reply by MatteoGioia** [reply]
Hi @rusty1s and thanks for taking the time to answer :) 

Yes it would be great, the main issue with the code in the PR mentioned above is that the augmentation code is really specific for MeshCNN. I tried to readapt some of the augmentations (and they work) but it would be nice to have them as transforms in a DataLoader for ease of use.

**Reply by rusty1s** [reply]
Yes, I think that's one of the reasons that PR isn't merged yet. Please feel free to bring these augmentations to PyG in a more general and PyG-way :)
